Sources/Articles/Documentation.docc/2024/swiftpm-snippets/SwiftPM Snippets.md

```diff
@@ -0,0 +1,264 @@
+
+Some documentation engines such as Unidoc support referencing Snippets by `id`.
+
+@Code(name: "My article.md", file: "My article (1).md.txt")
+
+>   Important:
+>   DocC does not currently support referencing Snippets by ID. Instead, you must use the fully-qualified `path` syntax to reference a Snippet.
+
+You should now have a project layout that looks like this:
+
+```
+📂 swift-snippets
+├── 📂 Snippets
+│   └── 📄 SnippetsExample_I.swift
+├── 📂 Sources
+│   └── 📂 SnippetsExample
+│       ├── 📂 docs.docc
+│       │   └── 📄 My article.md
+│       └── 📄 anchor.swift
+├── 📄 Package.swift
+└── 📄 .gitignore
+```
+
+
+### Previewing Snippets with DocC
+
+Many developers find [DocC](https://github.com/apple/swift-docc) helpful for previewing documentation locally. To use DocC, add the [swift-docc-plugin](https://github.com/apple/swift-docc-plugin) to the package manifest.
+
+@Code(name: Package.swift, file: Manifest.3.swift)
+
+You can then launch DocC with the `preview-documentation` package subcommand:
+
+```bash
+$ swift package --disable-sandbox preview-documentation --target SnippetsExample
+```
+
+You can find the rendered article at [`http://localhost:8080/documentation/snippetsexample/my-article`](http://localhost:8080/documentation/snippetsexample/my-article).
+
+>   Warning:
+>   At the time of writing, there is a [known bug](https://github.com/apple/swift-docc/issues/944) causing DocC preview to crash if the target name contains special characters.
+
+
+## Using Snippet captions
+
+If a Snippet begins with contiguous line comments, those comments will be parsed as Markdown and treated as a Snippet caption. Try adding the following code to a Snippet named `SnippetsExample_II.swift`.
+
+@Code(name: "SnippetsExample_II.swift", file: SnippetsExample_II.swift)
+
+When embedded, it should look like this:
+
+---
+
+@Snippet(id: SnippetsExample_II)
+
+>   Note:
+>   Snippet captions cannot embed other Snippets.
+
+
+## Redacting parts of a Snippet
+
+You can redact portions of a Snippet using **slice directives**. A slice directive is a line comment token that starts with `snippet` followed by a dot and an identifier. The following identifiers have special meanings:
+
+| Identifier        | Behavior                                   |
+| ----------------- | ------------------------------------------ |
+| `snippet.hide`    | Hides the content following the directive. |
+| `snippet.end`     | Hides the content following the directive. |
```

Comment:
```suggestion
| `snippet.end`     | Ends the current slice. |
```

Reply:
this was repeated intentionally, from an implementation standpoint `snippet.end` and `snippet.hide` are identical.

Reply:
Ah ok, you are completely right about this. :)